lib/rmt/lockfile.rb

```diff
@@ -1,20 +1,24 @@
 class RMT::Lockfile
-  LOCKFILE_LOCATION = File.expand_path('../../tmp/rmt.lock', __dir__).freeze
+  LOCKFILE_LOCATION = '/run/lock/rmt/.lock'.freeze
   ExecutionLockedError = Class.new(StandardError)
 
   class << self
-    def create_file
-      if File.exist?(LOCKFILE_LOCATION)
-        raise RMT::Lockfile::ExecutionLockedError
-      else
-        dirname = File.dirname(LOCKFILE_LOCATION)
-        FileUtils.mkdir_p(dirname) unless File.directory?(dirname)
-        File.open(LOCKFILE_LOCATION, 'w') { |f| f.write(Process.pid) }
-      end
-    end
+    def lock
+      # https://ruby-doc.org/core-2.5.0/File.html#method-i-flock
+      File.open(RMT::Lockfile::LOCKFILE_LOCATION, File::RDWR | File::CREAT) do |f|
+        if f.flock(File::LOCK_EX | File::LOCK_NB)
+          f.flock(File::LOCK_UN) # unlock for writing the PID
```

Comment:
You don't need to unlock the file to write it!

Reply:
fixed